float64 vel_esq
float64 vel_dir